packages/react-hooks/src/useBlockInterval.ts

```diff
@@ -2,48 +2,29 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import type { ApiPromise } from '@polkadot/api';
+import type { BabeGenesisConfiguration } from '@polkadot/types/interfaces/babe';
 
 import { useMemo } from 'react';
 
-import { BN, BN_THOUSAND, BN_TWO, bnMin } from '@polkadot/util';
+import { BN } from '@polkadot/util';
 
 import { createNamedHook } from './createNamedHook.js';
 import { useApi } from './useApi.js';
-import { A_DAY } from './useBlocksPerDays.js';
-
-// Some chains incorrectly use these, i.e. it is set to values such as 0 or even 2
-// Use a low minimum validity threshold to check these against
-const THRESHOLD = BN_THOUSAND.div(BN_TWO);
-const DEFAULT_TIME = new BN(6_000);
-
-function calcInterval (api: ApiPromise): BN {
-  return bnMin(A_DAY, (
-    // Babe, e.g. Relay chains (Substrate defaults)
-    api.consts.babe?.expectedBlockTime ||
-    // POW, eg. Kulupu
-    api.consts.difficulty?.targetBlockTime ||
-    // Subspace
-    api.consts.subspace?.expectedBlockTime || (
-      // Check against threshold to determine value validity
-      api.consts.timestamp?.minimumPeriod.gte(THRESHOLD)
-        // Default minimum period config
-        ? api.consts.timestamp.minimumPeriod.mul(BN_TWO)
-        : api.query.parachainSystem
-          // default guess for a parachain
-          ? api.consts.aura?.slotDuration ?? DEFAULT_TIME.mul(BN_TWO)
-          // default guess for others
-          : DEFAULT_TIME
-    )
-  ));
-}
+import { useCall } from './useCall.js';
 
 function useBlockIntervalImpl (apiOverride?: ApiPromise | null): BN {
-  const { api } = useApi();
+  let { api } = useApi();
```

Comment:
We shouldn't reassign here, instead have a `const { api } = useApi()`, and then rename the api below to be `curApi` or something of the sort, so: `const curApi = apiOverride || api;`

That way there is less confusion.

Reply:
Done in [0ed9b77](https://github.com/polkadot-js/apps/pull/10457/commits/0ed9b779e3855f390409812b25075c1b0dec67a7)